### PR TITLE
projectile definition update

### PR DESCRIPTION
**Pull request by Kantanomo**
Rewriting the projectile tag definition to new standard.

### CODE REVIEW

File: xlive/Blam/Engine/items/projectile_definition.h [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	e_projectile_material_response_flags flags;
+	// Explaination("default result", "(if the potential result, below, fails to happen)")
+	e_projectile_material_response response;
+
+	// DO NOT USE
+	tag_reference old_effect; // effe
+
+	string_id material_name;
+
+	PAD(4);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/items/projectile_definition.h]: Don't use pad macro

File: xlive/Blam/Engine/items/projectile_definition.h [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+	e_projectile_material_response potential_response;
+	e_projectile_material_response_unit unit_response;
+	real32 chance_fraction;
+	real_bounds between;
+	real_bounds _and;
+	// DO NOT USE
+	tag_reference old_effect_2;
+	e_projectile_material_response_scale_effects scale_effects_by;
+	PAD(2);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/items/projectile_definition.h]: Don't use pad macro

File: xlive/Blam/Engine/items/projectile_definition.h [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+	tag_reference super_detonation; // effe
+
+	tag_reference super_detonation_damage; // jpt!
+
+	tag_reference detonation_sound; // snd!
+
+	e_damage_reporting_type damage_reporting_type;
+
+	PAD(3);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/items/projectile_definition.h]: Don't use pad macro

File: xlive/Blam/Engine/items/projectile_definition.h [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	// bullet's velocity when inflicting maximum damage
+	real32 initial_velocity;
+	// bullet's velocity when inflicting minimum damage
+	real32 final_velocity;
+
+	real_bounds guided_angular_velocity;
+
+	real_bounds acceleration_range;
+
+	PAD(4);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/items/projectile_definition.h]: Don't use pad macro

File: xlive/Blam/Engine/items/projectile_definition.h [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	_projectile_flag_oriented_along_velocity = FLAG(0),
+	_projectile_ai_must_use_ballistic_aiming = FLAG(1),
+	_projectile_detonation_max_time_if_attached = FLAG(2),
+	_projectile_has_super_combining_explosion = FLAG(3),
+	_projectile_damage_scales_based_on_distance = FLAG(4),
+	_projectile_travels_instantaneously = FLAG(5),
+	_projectile_steering_adjusts_orientation = FLAG(6),
+	_projectile_dont_noise_up_steering = FLAG(7),
+	_projectile_can_track_behind_itself = FLAG(8),
+	_projectile_ROBOTRON_STEERING = FLAG(9),
```
Comment (Berthalamew) [inline xlive/Blam/Engine/items/projectile_definition.h:46]: use lowercase

Comment (Kantanomo) [reply xlive/Blam/Engine/items/projectile_definition.h:46]: Bungie™

**Pull request merged**